toxcore/Messenger.c

```diff
@@ -2778,7 +2775,7 @@ static uint32_t friend_size(void)
     VALUE_MEMBER(userstatus);
     data += 3; // padding
     VALUE_MEMBER(friendrequest_nospam);
-    VALUE_MEMBER(last_seen_time);
+    ARRAY_MEMBER(last_seen_time);
```

Comment:
Maybe a stupid question. Why don't we use a `sizeof(struct...)`? Why these `VALUE_MEMBER` and `ARRAY_MEMBER` macros?

Reply:
Because there are different operations in different functions that I want to have exactly the same syntax. Consider this list of calls the "reflection metadata" for the struct, and the macro definitions the actual code that uses the metadata to do whatever it needs to do.